### PR TITLE
[BugFix] Fix l1 file size keep growing

### LINKED ISSUE #11868: L1 file size keep growing but the data size is far smaller than it
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...' with persistent index
2. INSERT INTO '....'
3. SELECT  '....'

### Real behavior (Required)
caeo1Q7kpV

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/starrocks/issues/11868 https://github.com/StarRocks/starrocks/issues/11721

## Problem Summary(Required) ：

For https://github.com/StarRocks/starrocks/issues/11868:

In branch-2.4, we support the varlen key of persistent index. But the estimate l1 size is not  accurate because the following two reasons:
1.  We use the total file size of previous l1 file as the real l1 data size
2.  We ignore the same key in l0 and l1, and calculate size twice

These two reasons will cause l1 file size keep growing but the real data size is far smaller than file size.

For https://github.com/StarRocks/starrocks/issues/11721:

`_total_kv_pairs_usage` of `SliceMutableIndex` is not update when load snapshot. So the `_total_kv_pairs_usage` is not accurate after load snapshot which leads to an error in the result of `dump_bound()`. And it may  lead to subsequent metadata errors.

So we will use the read data file size as the snapshot size to avoid potential errors.

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/common/config.h [hunk 1]
```diff
@@ -834,4 +834,6 @@ CONF_Int64(block_cache_disk_size, "21474836480"); // 20GB
 CONF_Int64(block_cache_block_size, "1048576");
 CONF_Int64(block_cache_mem_size, "2147483648"); // 2GB
 
+CONF_mInt64(kL0L1MergeRatio, "10");
```
Comment (decster) [inline be/src/common/config.h:837]: change name to follow convention

**Review by decster — DISMISSED** [review]

File: be/src/common/config.h (hunk 1 above)
Comment (sevev) [reply be/src/common/config.h]: > change name to follow convention

done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by decster — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
@mergify backport branch-2.4

**Pull request merged**